formats/json-jackson/src/main/java/io/cloudevents/jackson/CloudEventDeserializer.java

```diff
@@ -92,6 +92,9 @@ public <T extends CloudEventWriter<V>, V> V read(CloudEventWriterFactory<T, V> w
 
                 // Parse datacontenttype if any
                 String contentType = getOptionalStringNode(this.node, this.p, "datacontenttype");
+                if (contentType == null && this.node.has("data")) {
```

Comment:
I think this is aligned w/ the `JSON-format.md` file, section 3.1:

https://github.com/cloudevents/spec/blob/main/cloudevents/formats/json-format.md#31-handling-of-data

As for the non-binary / none base64 case it states:
> If the `datacontenttype` is unspecified, processing SHOULD proceed as if the
`datacontenttype` had been specified explicitly as `application/json`.

However the spec says _should_ not _must_.

Perhaps that's part of the issue?

Reply:
Yes, it can introduce dubious questioning as the one we are having in the Knative community. @duglin can you chime in?

Reply:
ping @duglin

Reply:
I'm not sure I'm understanding the question. Is it: why is it `SHOULD` and not a `MUST` in that part of the spec?  If so, I can't remember for sure, but I suspect it was done to allow for some out of band knowledge built into the event producer that allowed for it to "just know" what the real type is and take the appropriate serialization action. But w/o any extra knowledge things should assume/default to JSON.  Make any sense?